src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/location/LocationTest.java

```diff
@@ -0,0 +1,137 @@
+
+import io.restassured.response.Response;
+import io.zonky.test.db.AutoConfigureEmbeddedDatabase;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.pip.data.management.Application;
+import uk.gov.hmcts.reform.pip.data.management.models.location.Location;
+import uk.gov.hmcts.reform.pip.data.management.utils.FunctionalTestBase;
+import uk.gov.hmcts.reform.pip.data.management.utils.OAuthClient;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpHeaders.AUTHORIZATION;
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.hmcts.reform.pip.data.management.utils.TestUtil.BEARER;
+
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles(profiles = "functional")
+@AutoConfigureEmbeddedDatabase(type = AutoConfigureEmbeddedDatabase.DatabaseType.POSTGRES)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest(classes = {Application.class, OAuthClient.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class LocationTest extends FunctionalTestBase {
+
+    @Value("${system-admin-provenance-id}")
+    private String systemAdminProvenanceId;
+
+    private static final String LOCATION_LIST = "locationList";
+    private static final String TEST_LOCATION_NAME = "TestLocationOne";
+    private static final String TEST_LOCATION_ID = "2001";
+    private static final String TEST_JURISDICTION1 = "Magistrates";
+    private static final String TEST_JURISDICTION2 = "Family";
+    private static final String TEST_REGION = "North West";
+
+    private static final String BASE_LOCATIONS_URL = "/locations";
+    private static final String UPLOAD_LOCATIONS_URL = BASE_LOCATIONS_URL + "/upload";
+    private static final String GET_LOCATION_BY_ID_URL = BASE_LOCATIONS_URL + "/" + TEST_LOCATION_ID;
+    private static final String GET_LOCATION_BY_NAME_URL = BASE_LOCATIONS_URL + "/name/" + TEST_LOCATION_NAME
+        + "/language/ENGLISH";
+    private static final String GET_LOCATION_BY_REGION_AND_JURISDICTION = BASE_LOCATIONS_URL + "/filter";
+    private static final String DOWNLOAD_CSV_LOCATIONS_URL = BASE_LOCATIONS_URL + "/download/csv";
+
+    @Test
+    void locationControllerHappyPathTests() throws Exception {
+        Map<String, String> headerMap = new ConcurrentHashMap<>();
```

Comment:
It is cleaner to use Map.of() in one line when creating Map: 
Map<String, String> headerMap = Map.of(AUTHORIZATION, BEARER + accessToken);

Reply:
Used Map.of(